llvm/test/MC/ELF/nobits-non-zero-value.s

```diff
@@ -12,5 +12,8 @@
 # CHECK: {{.*}}.s:[[#@LINE+1]]:3: error: SHT_NOBITS section '.bss' cannot have instructions
   addb %al,(%rax)
 
+# CHECK: {{.*}}.s:[[#@LINE+1]]:11: warning: ignoring non-zero fill value in SHT_NOBITS section '.bss'
+.align 4, 42
+
```

Comment:
Add check lines for `--no-warn` / `--fatal-warnings`?

Reply:
> Add check lines for `--no-warn` / `--fatal-warnings`?

Can you walk me through why we might want that, please? I'd have guessed that if we properly emit a warning and the diagnostic infrastructure is working as intended then we wouldn't need to test such interactions for all kinds of warnings.

Reply:
`Warning()` returns true (indicating an error) only in presense of `--fatal-warnings`. We might want to add the `--fatal-warnings` check to test that we didn't forget `ReturnVal |=` part. (However, it appears that omitting it does not change the behavior because top-level loop not only checks for the return value, but also for what `hasPendingErrors()` returns.)
Regarding `--no-warn`, I tend to agree with you.

Reply:
I added a RUN with `--fatal-warnings`. Let me know if it addresses your concerns properly.

Reply:
This check line would match if `Warning` was replaced with `Error`. I expected something like:

```
# RUN: not llvm-mc ... | FileCheck %s
# RUN: not llvm-mc --fatal-warnings ... | FileCheck --check-prefix=FATAL %s
...
# CHECK: ... warning: ignoring ...
# FATAL: ... error: ignoring...
```

I grepped mc tests and found only 4 tests that excercise `--fatal-warnings`, so I guess you can just drop the last commit and go ahead...